src/jupytext/jupytext.py

```diff
@@ -232,12 +238,17 @@ def writes(self, nb, metadata=None, **kwargs):
         if self.fmt.get(
             "format_name"
         ) == MYST_FORMAT_NAME or self.ext in myst_extensions(no_md=True):
-            pygments_lexer = metadata.get("language_info", {}).get(
+            default_lexer = metadata.get("language_info", {}).get(
                 "pygments_lexer", None
             )
+            if default_lexer is None:
+                language = metadata.get("kernelspec", {}).get(
+                    "language"
+                ) or metadata.get("jupytext", {}).get("main_language")
```

Comment:
Notebooks constructed from minimalist MyST notebooks don't have any explicit metadata. When reading the MyST notebooks we will now parse the lexer and identify the notebook language from them.

Reply:
Changed to `jupytext.default_lexer` which unlike the above will work for all languages